source/agora/consensus/ValidatorSet.d

```diff
@@ -546,6 +547,44 @@ public class ValidatorSet
 
         return true;
     }
+
+    /***************************************************************************
+
+        Restore validators' information from blocks
+
+        Params:
+            blocks = blocks to get validators' information
+            finder = the delegate to find UTXOs with
+
+    ***************************************************************************/
+
+    public void restoreValidators (const(Block)[] blocks,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        foreach (const ref block; blocks)
+        {
+            if (block.header.enrollments.length > 0)
+            {
+                foreach (const ref enroll; block.header.enrollments)
+                {
+                    if (!this.hasEnrollment(enroll.utxo_key))
+                    {
+                        if (!this.add(block.header.height, finder, enroll))
+                            assert(0);
+                    }
+                    try
```

Comment:
Why the `try` if `updateEnrolledHeight` is `nothrow` ?

Reply:
It's my mistake. It will correct it.